packages/edit-site/src/index.js

```diff
@@ -158,6 +159,7 @@ export function initializeEditor( id, settings ) {
 
 	dispatch( blocksStore ).__experimentalReapplyBlockTypeFilters();
 	registerCoreBlocks();
+	registerLegacyWidgetBlock();
```

Comment:
Let's move the Legacy Widget block to `@wordpress/block-library` so that `registerCoreBlocks()` will register it.

It was kept in `@wordpress/widgets` because only the widgets editor and widgets customizer (2/4 editors) used it, but this PR makes it so that the site editor and post editor (which also supports template parts) uses it (4/4 editors).

Reply:
Oh, also, please test whether Legacy Widget block actually works in these new contexts. From memory I think that Legacy Widget requires that the `widget.php` assets are enqueued into the page.

Reply:
Do you mind if we do this as a follow-up? I don't think it's a blocker for this feature.

Reply:
Sure.

Reply:
Currently, I think we only need the Legacy Widgets block only for two reasons:

1. The registry recognizes it using the `createBlock` method.
2. To access the transformations.

Basically, register this block only, so it's only available in the registry. Have we done anything similar in the past?

Reply:
Right, yeah. I don't see a way around registering Legacy Widget. It means it will show up in the inserter though. Can we hide it?